File: src/main/java/org/zkoss/openlayers/layer/OLGoogle.java
/* OLGoogleV3.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Jun 18, 2012 4:59:11 PM , Created by jumperchen
}}IS_NOTE

Copyright (C) 2012 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
}}IS_RIGHT
*/
package org.zkoss.openlayers.layer;

import org.zkoss.lang.Objects;
import org.zkoss.zk.ui.AbstractComponent;

/**
 * @author jumperchen
 *
 */
public class OLGoogle extends OLLayer {
	private String _type;
	public OLGoogle() {}
	public OLGoogle(String type) {
		_type = type;
	}
	public void setType(String type) {
		if (!Objects.equals(_type, type)) {
			_type = type;
			smartUpdate("type", _type);
		}
	}
	//super//
	protected void renderProperties(org.zkoss.zk.ui.sys.ContentRenderer renderer)
	throws java.io.IOException {
		super.renderProperties(renderer);

		render(renderer, "type", _type);
	}

}

File: src/main/resources/web/js/openlayers/layer/OLGoogle.js
/** OLGoogle.js

	Purpose:
		
	Description:
		
	History:
		Mon, Jun 18, 2012  5:14:20 PM, Created by jumperchen

Copyright (C) 2012 Potix Corporation. All Rights Reserved.
*/
(function () {
openlayers.layer.OLGoogle = zk.$extends(openlayers.layer.OLLayer, {
	$define: {
		type: null
	},
	toOLWidget: function () {
		return new OpenLayers.Layer.Google(this.uuid, this.getOptions());
	},
	getOptions: function () {
		var options = this.$supers('getOptions', arguments);
		if (this._type) {
			options.type = $eval(this._type);
		}
		return options;
	}
});
})();